commands/core/core.go

```diff
@@ -0,0 +1,37 @@
+package core
+
+import (
+	"github.com/arduino/arduino-cli/arduino/cores"
+	rpc "github.com/arduino/arduino-cli/rpc/commands"
+)
+
+// platformReleaseToRPC converts our internal structure to the RPC structure.
+// Note: this function does not touch the "Installed" field of rpc.Platform as it's not always clear that the
+//       platformRelease we're currently converting is actually installed.
+func platformReleaseToRPC(platformRelease *cores.PlatformRelease) *rpc.Platform {
+	boards := make([]*rpc.Board, len(platformRelease.Boards))
+	i := 0
+	for _, b := range platformRelease.Boards {
```

Comment:
let's do `for i, b := range platformRelease.Boards` and let `range` deal with the index increments

Reply:
That would be my preferred way, too. `Boards` is a `map[string]..` though, thus `i` would not be numeric index here unfortunately.

Reply:
oops my bad, didn't check what's been iterated here, scratch my comment!